### PR TITLE
Schedule generation

### PR DESCRIPTION
**Pull request by Ahsen19**
Given a date, a schedule object will be returned containing all the chores due on that date. More information included in [CK-81](https://keepers.atlassian.net/browse/CK-81)

There is a possibility of small bugs coming up once the frontend implements:
- `lastCheckedOff` and `nextOccurrence` params to the chore [CK-94](https://keepers.atlassian.net/browse/CK-94)
- requests for schedules

### CODE REVIEW

File: server/controller/schedule.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        // get the user that is requesting the schedule
+        const userID = req.user.id;
+        const user = await User.findOne({ _id: userID }); // search for the user
+        const choreList = user.chores;
+        
+        console.log("Printing chore ids for this user")
+
+        for (const choreId of choreList) {
+            // console.log(choreId);
+            const chore = await Chore.findOne({ _id: choreId });
```
Comment (JoshuaDueck) [inline server/controller/schedule.js:40]: I don't think we need to be running a query against the DB for every chore in the list. `Chore.find( { _id: { $in: user.chores } });` automatically finds all chores with an ID in the user's list of chore IDs.

I think you should be able to put that line outside of the loop, populating some variable with the chore objects so we can run one query to get all the chores, then do whatever needs to be done.

**Review by JoshuaDueck — CHANGES_REQUESTED** [review]
I didn't think we were going to be going forward with all this schedule logic. It doesn't hurt to have this logic in here I guess, but we may not use it.

File: server/controller/schedule.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+            console.log(`Repeat in days: ${repeatInDays}`);
+
+            // calculating difference in days between nextOccurrence and the requested schedule date
+            const diffInMs = timeframeInMs - chore.nextOccurrence;
+            const diffInDays = Math.floor(diffInMs / (1000 * 60 * 60 * 24));
+
+            // now if the chore is scheduled for the given date, add it to the schedule list
+            if (diffInDays % repeatInDays == 0) {
+                requestedSchedule.push(chore);
+            }
```
[the remark this answers is not in the document]
Comment (Ahsen19) [reply server/controller/schedule.js:67]: Done

File: server/controller/schedule.js (hunk 1 above)
Comment (Ahsen19) [reply server/controller/schedule.js]: Done

**Review by Amikz — APPROVED** [review]
I thought we were going to simplify the schedule logic by having a variable for the last unchecked date and one for the next occurrence in the future (timewise, not whether it was checked off), so that we could just return a chore for the schedule if next == the current day, and just update next by adding the interval to it if it's in the past compared to the current day. Maybe I misunderstood though. 

But this is all fine to keep since it's already done, and it would allow us to display more than just the current date. We should probably meet to decide what we want the schedule page to actually look like.

**Pull request merged**